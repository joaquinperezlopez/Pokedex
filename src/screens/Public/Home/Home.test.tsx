import { i18n } from '@i18n/index';
import { describe, expect, jest, test } from '@jest/globals';
import { PublicStackNavigationProps } from '@navigation/public/public.navigator.types';
import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import HomeScreen from './Home.screen';

const mockNavigationProps = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    reset: jest.fn(),
    setParams: jest.fn(),
    dispatch: jest.fn(),
    addListener: jest.fn(),
    removeListener: jest.fn(),
    canGoBack: jest.fn(),
    isFocused: jest.fn(),
    dangerouslyGetParent: jest.fn(),
    dangerouslyGetState: jest.fn(),
  },
  route: {
    key: 'Home',
    name: 'Home',
    params: {},
  },
} as unknown as PublicStackNavigationProps<'Home'>;

describe('HomeScreen', () => {
  test('renders without crashing', () => {
    const { toJSON } = render(<HomeScreen {...mockNavigationProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('calls onPressEmailLogin when PrimaryButton is pressed', () => {
    const { getByText } = render(<HomeScreen {...mockNavigationProps} />);
    const emailLoginButton = getByText('screens.home.emailLogin');
    fireEvent.press(emailLoginButton);
    expect(mockNavigationProps.navigation.navigate).toHaveBeenCalledWith(
      'Login',
      {
        email: '',
      }
    );
  });

  test('calls onSignUpPress when SecondaryButton is pressed', () => {
    const { getByText } = render(<HomeScreen {...mockNavigationProps} />);
    const signUpButton = getByText('screens.home.signUp');
    fireEvent.press(signUpButton);
    expect(mockNavigationProps.navigation.navigate).toHaveBeenCalledWith(
      'SignUp'
    );
  });

  test('calls onSelectSpanish when Spanish flag is pressed', () => {
    const { getByText } = render(<HomeScreen {...mockNavigationProps} />);
    const spanishFlag = getByText('ðŸ‡ªðŸ‡¸'); // Replace this with the actual Spanish flag text
    fireEvent.press(spanishFlag);
    // Add expectations for the changes that should occur after pressing the Spanish flag
    expect(i18n.locale).toBe('es');
  });

  test('calls onSelectEnglish when English flag is pressed', () => {
    const { getByText } = render(<HomeScreen {...mockNavigationProps} />);
    const englishFlag = getByText('ðŸ‡¬ðŸ‡§'); // Replace this with the actual English flag text
    fireEvent.press(englishFlag);
    expect(i18n.locale).toBe('en');
  });
});
