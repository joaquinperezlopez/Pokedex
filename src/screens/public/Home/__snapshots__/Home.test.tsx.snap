// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders without crashing 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#DC0A2D",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "gap": 16,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/images/pokeball.png",
        }
      }
    />
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 32,
          "fontWeight": "bold",
        }
      }
    >
      appName
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "stretch",
        "flex": 1,
        "gap": 20,
        "justifyContent": "center",
        "width": "80%",
      }
    }
  >
    <View
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#DC0A2D",
            "borderColor": "#E0E0E0",
            "borderRadius": 6,
            "borderWidth": 1,
            "justifyContent": "center",
            "padding": 10,
          },
          undefined,
        ]
      }
      testID="primaryButtonscreens.home.emailLogin"
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        screens.home.emailLogin
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        screens.home.signUp
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "gap": 16,
        "padding": 10,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 25,
          "justifyContent": "center",
          "paddingBottom": 20,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 32,
              "fontWeight": "bold",
            }
          }
        >
          ðŸ‡ªðŸ‡¸
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 32,
              "fontWeight": "bold",
            }
          }
        >
          ðŸ‡¬ðŸ‡§
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
